# Genomics report
Here is the final report of my analysis. 

You can find the workflow used to obtain the result in the Workflow.md file
## Abstarct 
In this report I present my analysis of the assigned cases using the different tools presented
during the lesson. The aim is making a molecular diagnosis of a rare genetic disorder of the
5 individuals. The methodology is slightly the same for each trios, the difference is in the
choice of segregation path, depending on the type of disease: autosomal dominant or
autosomal recessive.
For each case I checked the quality of the genome sequence data and then I created the vcf
file to analyze with VEP and make a conclusion. For one case I reported the quality
assessment using Qualimap and MultiQC and the analysis of one disease causing variant
with UCSC genome browser.

## Table of resultant disease 

| Case  | Chr |Position|Ref|Alt| Type| Phenotype
| ------------- | ------------- | ----- | ----- | ---- | ---- | --- |
| 1677 | 16  | 2144164- 2144166 | ACG | AGCTGGC | frameshift_variant | Autosomal dominant polycystic kidney disease
| 1689  | 16  | 51174854-51| TTCT| TT|frameshift_variant | Townes-Brocks syndrome 1 |
| 1832| 16| 89341551-89341551| G | A | stop_gained| KBG syndrome|
|1743| 16|56535271-56535273| AAG| ACCATTACAG|frameshift_variant| Bardet-Biedl Syndrome|
|1796|16|89347502-89347506| AGGAA|AGAA|frameshift_variant| KBG syndrome| 

## Quality check 
After the production of fastQC report and Qualimap histograms, I decided to analyze each
case with MultiQC to evaluate each metrics at once.
I started my analysis looking at the general statistic:
* % Aligned ~ 99.9% which is very good
* % Duplicate is not high (<10%), since I am considering exome sequence I expect a moderate level of duplication.
* %CG usually is around 40% but my sequences are represent exomes so higher level of this parameter is acceptable.

This is a first approach to state the quality of the sequences, it’s necessary to consider other factors to assure these conclusions.
Because of this i checked the other graph:
* Coverage histograms are consistent with the exome sequences: the samples are composed of high coverage regions.
* FastQC report gives acceptable results for the various analyses.Note that I ignore the CG content on the child case because usually FastQC gives false negatives in the exome cases (from literature).

I double checked the results analyzing the mapping quality histogram on Qualimap and I observed a high number. This means that the alignments generated by bowtie2 are quite good.
I also checked the other cases obtaining similar results, so I can conclude that the data are quite good and I can go on with the analysis.

You can find an example of my report in multiqc_report.html

## VEP analysis
After producing the filtered vcf files for each case, I loaded them into VEP in order to do
variant annotation.
I set some parameters in VEP (GRCh37) on the various vcf files, with some exception as I
presented below:
* Transcripts: Refseq
* Allele Freq data: 1000 genomes global and gnomAD
* Additional annotations : phenotypes + Mastermind
* Pathogenicity predictions: prediction + score (SIFT+polyphen)

Then I applied a different approach for autosomal dominant cases and the autosomal recessive one, as they require different filters to do variant annotation.
The explanation of the choices on the filters is described below.

### AD cases
Since the autosomal dominant cases are considering de novo mutation (also rare Mendelian disease), I decided to set the filter “IMPACT is HIGH” and then I checked the allele frequency to be low or equal to “-”. This is because the novo mutations are characterized by a strong impact that causes protein loss of function or mediated decay. 

De novo mutations are usually very rare and so it’s necessary to ensure that the allele frequency is very low (note that in VEP AF equal to “-” means < 0.001).
In these cases it was impossible to apply SIFT and Polyphen because they can’t be used in cases like frameshift mutations or for mutations in regulatory regions.

After this I selected the row with associated phenotype to do my diagnosis.

### AR case
For the autosomal recessive case, at first, I considered different filters. We are considering a rare Mendelian disease also in this case, so I set the filters in a way that ensures to find high damaging diseases that are very rare.

At first, I didn’t consider the rows with low or modifier impact. Then I added filters on SIFT and Polyphen to detect damaging mutations, but I couldn't find anything relevant. So I decided to add CADD in the analysis to see if the mutation is caused by a single nucleotide variant or insertions/deletions, but also in this case I didn't find any associated phenotype
with the required characteristics.

So I decided to follow a straightforward approach as done for AD cases: I set “IMPACT is "HIGH” and filtered the rows with an associated phenotype, checking for lower allele frequencies (also in this case the acceptable rows has “AF is - ”).

This allowed me to find a diagnosis consistent with the characteristic of a rare mendelian disease

## Coverage track with UCSC Genome Browser (case 1689)
![UCSCresult](https://github.com/MajerIrene/Autosomic-Diagnosis/blob/main/UCSCresult.png)

This image is taken from UCSC Genome Browser and shows the disease-causing variant.
For the analysis I uploaded the trio’s bedgraph files obtained with bedtools genomecov (as
seen before in the command section). Then I also uploaded the final vcf of my analysis to
show the disease causing variant.
In the picture is it possible to analyze case 1689: here I found a frameshift in position
51174854-51174857 with loss of bases T and C, as represented in this image with the gap in
the child’s track with respect to mother and father.

## Conclusions 
Before stating a diagnosis I double checked with the literature of each disease, comparing
my prediction with the genomic explanation of the associated disease.
In case 1677 I found that the disease associated is Autosomal dominant polycystic kidney
disease, a mutation in the region 16p13.3. In VEP I found that the gene symbol related was
PKD1 as shown in literature and the position found (2144164-2144166) corresponds to the
correct region.
I made a similar workflow for the other cases and I found reasonable proof to confirm what i
stated with my analysis using VEP.
At the end of this work I can conclude that the workflow seen during classes led me to make
diagnosis that are supported by real data, taken by literature about the different diseases.
